### PR TITLE
Pro 5297 cache fix

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR updates the cache and login modules. It was needed to allow for the cache-redis extension to function. Before this PR the number of login attempts was cleared by deleting the cached value from the MongoDB directly. As the cache-redis extension doesn't use MongoDB for caching, this led to an error. A new `delete` function was added to the cache module and the login module `clearLoginAttempts` method was altered to this new method. Closes PRO-5297.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly
1. Spin-up a project
2. Confirm that login proceeds without error.
## What kind of change does this PR introduce?
*(Check at least one)*

- [X] Bug fix
- [X] New feature
- [X] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/cache/index.js [hunk 1]
```diff
@@ -79,6 +79,15 @@ module.exports = {
         return self.cacheCollection.removeMany({ namespace });
       },
 
+      // Delete a single key and value from the cache.
+
+      async delete(username, namespace) {
```
Comment (boutell) [inline modules/@apostrophecms/cache/index.js:84]: Be consistent, namespace first

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/cache/index.js (hunk 1 above)
Comment (BoDonkey) [reply modules/@apostrophecms/cache/index.js:84]: Fixed

File: modules/@apostrophecms/login/index.js [hunk 2]
```diff
@@ -889,8 +889,8 @@ module.exports = {
         };
       },
 
-      async clearLoginAttempts (username, namespace = loginAttemptsNamespace) {
-        await self.apos.cache.delete(username, namespace);
+      async clearLoginAttempts (namespace = loginAttemptsNamespace, username) {
```
Comment (boutell) [inline modules/@apostrophecms/login/index.js:892]: Defaults for the first argument don't work if there is a second argument, which is required here

Also, tests are failing

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/login/index.js (hunk 2 above)
Comment (boutell) [reply modules/@apostrophecms/login/index.js:892]: I'm wrong about defaults not being supported for first arguments of several, if you pass undefined they are used

**Review by boutell — APPROVED** [review]

**Pull request merged**